### DISCUSSION TITLE
How to get vimdiff to highlight changes when using `jj split`?

### DISCUSSION THREAD

**Discussion by f0lie** [jj-vcs/jj#1245] [Q&A] [answered]
I am trying to split a commit. Since meld is broken on WSL2 (for me at least), I am using `vimdiff` instead. The problem is that I can't tell where my changes are when I am trying to split a commit. On the right and left sides, it's all just black-and-white text.

**Comment by f0lie** [comment, marked as answer]
https://gist.github.com/ilyagr/5d6339fb7dac5e7ab06fe1561ec62d45

glen in the discord very helpfully pointed out this link. If you follow this guide, `jj split` highlights the differences.

**Reply by chooglen** [reply]
The link came from https://github.com/martinvonz/jj/blob/main/docs/config.md. Thanks @ilyagr for the pointers.

I wonder what the underlying issue is. Does vimdiff have poor directory-diffing support?

**Reply by f0lie** [reply]
Looking at the shell file, it seems like it's using the plugin `DirDiff` to enable "recursive" `vimdiff` on two dictionaries.

So it does seem like `vimdiff` is geared towards diffing only two files. Although, I don't use `vimdiff` enough to say that for certain.

Note that I only have a single file that I was trying to split. Even when I open the same file on both sides, it still didn't highlight the results.

**Reply by ilyagr** [reply]
> Does vimdiff have poor directory-diffing support?

It has no such support at all. We could work around that in `jj` by providing pairs of files to the difftool, but that requires thinking about how to specify that in configuration.

I will also update that gist with a hacky way to open each pair of files in a separate tab (using another Vim plugin).

**Reply by ilyagr** [reply]
Actually, I would now recommend https://github.com/balki/vimtabdiff, which is a python script rather than a Vim plugin.

I will at some point include that in https://gist.github.com/ilyagr/5d6339fb7dac5e7ab06fe1561ec62d45 in addition to DirDiff that it already talks about. (DirDiff would be better for diffs with more than a few changed files, but apart from that I prefer vimtabdiff).
